### DISCUSSION TITLE
ABB Gofa CRB15000

### DISCUSSION THREAD

**Discussion by aimforkim** [visose/Robots#218] [Q&A] [answered]
Hello, 

I am trying to create an xml file for ABB Gofa CRB15000. There is an issue with the last axis (or maybe the last two axis), as I am unable to move the 6th axis plane up by 80mm along world-axis-Z, or what seems to be the 'a' value of the 5th axis in the xml file. Whatever value I put on the 5th axis does not seem to make any changes. 
![image](https://user-images.githubusercontent.com/96722124/220090635-eb996911-3046-470e-99b1-e7c2efada513.png)

I read the discussion for Staubli TX60 (https://github.com/visose/Robots/discussions/103) and Ufactory Lite6 (https://github.com/visose/Robots/discussions/214). 

I tried to copy from the TX60 xml file, but the TX60 file as well as my modified one for the Gofa are not recognized by my Robots plugin (which I've installed again a month ago). I've attached the zip file of the Gofa model and the xml file in case you could take a look. Any help would be hugely appreciated. Thank you! 
 
[Gofa.zip](https://github.com/visose/Robots/files/10783775/Gofa.zip)

**Comment by Paultn12** [comment]
Hi Aimforkim,

Did you manage to complete the xml file for the GoFa CRB 15000?

Just before I make a start myself..

Thank you,
Paul.

**Reply by barrakdarweesh** [reply]
Hello! 

I'm having the same issue figuring out the correct a and d values. Any assistance would be appreciated.

**Comment by visose** [comment, marked as answer]
Hi, this robot model does not have a spherical wrist like the other ABB industrial robots. To make this work in the plugin will require changes to the code base to use a different inverse kinematics solver for this model.

Forward kinematics with the correct DH parameters, as well as the RAPID code generation should work without changes.

**Reply by twisharaja** [reply]
Hello there,

Currently working on a project with the CRB 15000, have there been any updates on this?

Thanks,
Twisha

**Reply by visose** [reply]
Hi, no there are currently no plans to add support for this robot. This will only happen if the work is sponsored. If you are interested in sponsorings the work involved in adding support for this type of robot, you can email me at vicente@visose.com.
